org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.services.format;
+
+import java.util.List;
+
+import org.eclipse.lemminx.dom.DOMComment;
+import org.eclipse.lsp4j.TextEdit;
+
+/**
+ * DOM text formatter.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class DOMCommentFormatter {
+
+	private final XMLFormatterDocumentNew formatterDocument;
+
+	public DOMCommentFormatter(XMLFormatterDocumentNew formatterDocument) {
+		this.formatterDocument = formatterDocument;
+	}
+
+	public void formatComment(DOMComment commentNode, XMLFormattingConstraints parentConstraints,
+			List<TextEdit> edits) {
+
+		// Check the comment is closed properly
+		if (commentNode.getEnd() == commentNode.getEndContent()) {
+			return;
+		}
+		FormatElementCategory formatElementCategory = parentConstraints.getFormatElementCategory();
```

Comment:
I think comment text should not be formatted acording the FormatElementCategory

Please remove the following  ode which formats comment text according FormatElementCategory